pilot/lib/pilot/build_manager.rb

```diff
@@ -253,9 +253,11 @@ def update_beta_app_meta(options, build)
         end
       end
 
-      update_build_beta_details(build, {
-        auto_notify_enabled: options[:notify_external_testers]
-      })
+      if options[:notify_external_testers]
+        update_build_beta_details(build, {
+          auto_notify_enabled: options[:notify_external_testers]
+        })
+      end
```

Comment:
@acasademont It turns out that `auto_notify_enabled` is defaulted to `true` so this logic will actually always notify even when `notify_external_testers` is set to `false`

One thing we _could_ do is 👇  but this makes a pretty big assumption about how App Store Connect API works and will break if we change it 🤔 

_Maybe_ we end up making `:notify_external_testers` optional and only run this request if its explicitly set? This would work because it would apply App Store Connect APIs default but `true` and `false` would do exactly what its supposed to.

Thoughts?

```
      unless options[:notify_external_testers]
        update_build_beta_details(build, {
          auto_notify_enabled: options[:notify_external_testers]
        })
      end
```

Reply:
Hi @joshdholtz , I'm afraid I don't really understand this:

> @acasademont It turns out that `auto_notify_enabled` is defaulted to `true` so this logic will actually always notify even when `notify_external_testers` is set to `false`

If `notify_external_testers` is explicitly set false the API call  won't be made (i've used it quite a lot already in my dev environment)-
> 
> One thing we _could_ do is 👇 but this makes a pretty big assumption about how App Store Connect API works and will break if we change it 🤔
> 
> _Maybe_ we end up making `:notify_external_testers` optional and only run this request if its explicitly set? This would work because it would apply App Store Connect APIs default but `true` and `false` would do exactly what its supposed to.
> 
> Thoughts?
> 
> ```
>       unless options[:notify_external_testers]
>         update_build_beta_details(build, {
>           auto_notify_enabled: options[:notify_external_testers]
>         })
>       end
> ```

This is precisely what I wanted to avoid using the current default value, it needs to be explicitly set to `false`.

Reply:
@acasademont  Sorry, I should have been more clear!

```rb
 if options[:notify_external_testers]
```

The current code in the PR has this ☝️ `options[:notify_external_testers]` is a boolean (so either `true` or `false`). So this means that `update_build_beta_details` will only get called if `options[:notify_external_testers]` is set to `true`

This is problematic because App Store Connect (not _fastlane_) defaults the “notifying of external testers” to `true`. So...

If `options[:notify_external_testers]` is set to `false`, this if statement won’t get executed and App Store Connect will still notify external testers because it itself will keep it default to `true`.

My thoughts on a solution would be to:
1) Set the `notify_external_testers` to optional
2) Change the if statement to `unless options[:notify_external_testers].nil?`

So with this, `update_build_beta_details` will only get called if explicitly set and would retain backwards compatibility because App Store Connect defaults notifying external testers to true 😊

Reply:
aaah, gotcha, sorry for that!! Then, yes, definititely your suggestion is the way to go. And although it's a BC-break in terms of the fastlane API, it won't have any effect in terms of the actual notification to external testers if the App Store default is to notify the external testers.

Should I work on those changes?

Reply:
@acasademont I don’t think its a _full_ break since the option defaults to `true` right now and App Store Connect currently also defaults it to `true` 🤷‍♂️  But it _could_ break if App Store Connect changes it but at least it can be explicitly set with ease 😊 

If you want to make this change that would be great! I’d be happy to test it out again ❤️

Reply:
@joshdholtz done! thank you :D